index.js

```diff
@@ -57,21 +57,43 @@ async function publishResults (results) {
     throw err
   }
 
+  function annotateMessage ({ line, severity, ruleId, message }, path) {
+    const annotationLevel = {
+      1: 'warning',
+      2: 'failure'
+    }[severity]
+    return {
+      path,
+      start_line: line,
+      end_line: line,
+      annotation_level: annotationLevel,
+      message: `[${ruleId}] ${message}`
+    }
+  }
+
+  function clubMessagesOnSameLine (messages) {
+    const messagesClubbedByLine = {}
+    messages.forEach(message => {
+      messagesClubbedByLine[message.line] =
+        messagesClubbedByLine[message.line] || []
+      messagesClubbedByLine[message.line].push(message)
+    })
+    return messagesClubbedByLine
+  }
+
   function toAnnotations ({ filePath, messages }) {
     const path = filePath.substr(GITHUB_WORKSPACE.length + 1)
-    return messages.map(({ line, severity, ruleId, message }) => {
-      const annotationLevel = {
-        1: 'warning',
-        2: 'failure'
-      }[severity]
-      return {
-        path,
-        start_line: line,
-        end_line: line,
-        annotation_level: annotationLevel,
-        message: `[${ruleId}] ${message}`
-      }
-    })
+    const messagesClubbedByLine = clubMessagesOnSameLine(messages)
+    return Object.keys(messagesClubbedByLine).reduce(
+      (prevAnnotations, lineNumber) => {
+        return prevAnnotations.concat(
+          messagesClubbedByLine[lineNumber].map(message =>
+            annotateMessage(message, path)
```

Comment:
I think what I'm looking for is a little different. With this approach, it still puts multiple annotations on each line that has multiple issues. The goal is to have at most one annotation for each line. If there are multiple errors on one line, it should be turned into one annotation that has all the messages, like
```
[no-unused-var] x is unused
[prefer-const] x is not changed, use const instead
```

So each entry in `messagesClubbedByLine` should only result in _one_ `annotateMessage()` call, with all the different entries combined into one.